src/components/interestedPeople/SearchAndFilter.tsx

```diff
@@ -0,0 +1,142 @@
+"use client"
+import React, { useState, useRef, useEffect } from 'react';
+import { FiSearch, FiCalendar, FiTag } from 'react-icons/fi';
+import { CgChevronDown } from "react-icons/cg";
+import { SearchAndFilterProps } from '@/types/table';
+
+
+
+const dateRangeOptions = [
+    { label: 'Today', value: 'today' },
+    { label: 'Last 7 days', value: 'last7days' },
+    { label: 'Last 30 days', value: 'last30days' },
+    { label: 'Last 90 days', value: 'last90days' },
+    { label: 'Custom range', value: 'custom' }
+];
+
+const statusOptions = [
+    { label: 'All', value: '' },
+    { label: 'Pending', value: 'pending' },
+    { label: 'Accepted', value: 'accepted' },
+    { label: 'Rejected', value: 'rejected' },
+];
+
+export default function SearchAndFilter({
```

Comment:
Pass all this to a custom hook and call it here
This way it looks more organized and there is only the html